src/lib/y2security/selinux.rb

```diff
@@ -145,21 +220,39 @@ def configurable?
     GETENFORCE_PATH = "/usr/sbin/getenforce".freeze
     private_constant :GETENFORCE_PATH
 
-    # Find SELinux mode by given value
+    # Returns the values for the SELinux setting from the product features
     #
-    # @note using nil means to set SELinux mode as disabled.
+    # @return [Hash{Symbol => String, Boolean, nil}] e.g., { mode: "enforcing", configurable: false }
+    #   a hash holding the defined SELinux options in the control file;
+    #   an empty object if no settings are defined
+    def product_feature_settings
+      @product_feature_settings unless @product_feature_settings.nil?
+
+      settings = Yast::ProductFeatures.GetFeature("globals", "selinux")
+      settings = {} if settings.empty?
+      settings.transform_keys!(&:to_sym)
+
+      @product_feature_settings = settings
+    end
+
+    # Returns a CFA::Selinux object for handling the config file
     #
-    # @param id [SelinuxConfig::Mode, String, Symbol, nil] a SELinux mode or its identifier
-    # @return [Mode] the SelinuxConfig::Mode by given id or disabled is none found or nil was given
-    def find_mode(id)
-      found_mode = Mode.find(id)
+    # @return [CFA::Selinux]
+    def config_file
+      return @config_file if @config_file
 
-      if found_mode.nil?
-        log.info("Requested SELinux mode `#{id}` not found. Falling back to :disabled.")
-        found_mode = Mode.find(:disabled)
-      end
+      @config_file = CFA::Selinux.load
```

Comment:
can be simple `@config_file ||= CFA::Selinux.load`

Reply:
also I think exception should be catched if file does not exists?

Reply:
Right, that method was written before adding the class method `.load` and then it was not fully adapted. Good catch, thanks!